types/v1/k8s_storage_class.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+Copyright 2017 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file deals with the labels, constants, defaults & utility
+// functions related to the structures of K8s StorageClass Kind
+package v1
+
+import (
+	"fmt"
+	"strings"
+)
+
+// SCProvision type deals with the keys associated with
+// volume provisioning. It is expected that these keys will have
+// corresponding values.
+type SCProvisionKey string
```

Comment:
Going by the above, we can call this is PolicyKey

Reply:
I am trying to de-link `provision` from `policy`. 
A policy will mapped/related to multiple provisioning properties.
e.g. 
ReplicaPlacementPolicy will map to one or more volume properties & corresponding values. These values will be very specific to the Policy.

NOTE: There may be chances of conflicting policies. A policy will conflict with another policy if both of them have one or more volume properties in common & these common properties' value differ based on policy.

What are your thoughts ?

Reply:
still a bit confusing. i read this file as .. there are different kinds of policies - provisioning policies, replica placement policies. 

is it fair to .. view policy will map to volume - provisioning properties?

then, we could differentiate between a policy and property?

Reply:
currently policy seems to map to one more more provisioning properties i.e. fields inside a Volume Struct. It might happens that profile can map to additional stuff as well. However to abstract the inner details from storage admin i.e the one who maintains storage class, I have put profile. In addition, these profile are similar to flags. Presence of profile implies enabling that profile.